src/shared/seccomp-util.c

```diff
@@ -2072,3 +2076,83 @@ uint32_t scmp_act_kill_process(void) {
 
         return SCMP_ACT_KILL; /* same as SCMP_ACT_KILL_THREAD */
 }
+
+int safe_seccomp_load(scmp_filter_ctx c) {
+        _cleanup_free_ struct sock_filter *filter = NULL;
+        _cleanup_close_ int fd = -1;
+        size_t n_filter_blocks;
+        struct stat st;
+        uint32_t u;
+        ssize_t n;
+        int r;
+
+        /* So here's the problem: starting with libseccomp 2.5.0 seccomp_load() will always allocate a
+         * seccomp notifier fd for its filters, and store as global state (even if otherwise not used). We
+         * really don't like that for two reasons:
+         *
+         * 1. We install seccomp filters very late in our process startup logic, at a time where we don't
+         *    want global fds to stick around that could interfere with our fd closing spree and fd
+         *    reordering.
+         *
+         * 2. We want to install many seccomp filters, but since the notifier fd is a process-global concept
+         *    libseccomp currently only allows installation of a single filter, breaking compatibility with
+         *    previous library versions.
+         *
+         * Hence, let's sidestep the whole issue: on affected libseccomp versions, let's ask libseccomp for
+         * the raw BPF code, and let's then install it ourselves, without any notifier logic. It's not as
+         * pretty as we'd like, since libseccomp only allows exporting BPF records into files, but we can use
+         * memfds to work around that. */
+
+        assert(SCMP_VER_MAJOR >= 2); /* as per meson build dependency reqs */
```

Comment:
Is this a compile time check? There was no so-version bump between libseccomp 2.4 and 2.5, so we cannot assume that the same version is used during compilation and runtime. This check below needs to be runtime check.

Reply:
In case this wasn't clear: this would indeed mostly solve the issue in Fedora, where we could release an updated systemd version with the patch compiled against new libseccomp. But it doesn't solve the general case where people compile against old libseccomp and then upgrade it. (Like I did on my development machine...)

Reply:
it's a compile time check so far. will turn it into a runtime check.